src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySinkUtils.java

```diff
@@ -624,6 +630,10 @@ public static String generateUpdateUpsertQuery(Operation operation,
         return String.format(UPDATE_QUERY, destination, fieldsForUpdate, sourceTable, criteria);
       case UPSERT:
         String insertFields = "`" + String.join("`, `", tableFieldsList) + "`";
+        if (fieldsForUpdate.isEmpty()) {
```

Comment:
Add a comment explaining why we return one or the other

Reply:
Added comment